internal/controller/providers/psmdb/provider.go

```diff
@@ -139,6 +139,12 @@ func (p *Provider) Status(ctx context.Context) (everestv1alpha1.DatabaseClusterS
 		status.Status = everestv1alpha1.AppStateRestoring
 	}
 
+	// If the current version of the database is different from the version in
+	// the CR, an upgrade is pending or in progress.
+	if p.DB.Spec.Engine.Version != "" && psmdb.Status.MongoVersion != "" && p.DB.Spec.Engine.Version != psmdb.Status.MongoVersion {
```

Comment:
it really looks like a better approach move this logic to DB operators and reflect the `upgrading` state in db-operator's resource, isn't it?

Reply:
Yes, I also thought about that and was thinking about discussing this possibility with the cloud team in our next sync.